qa/sc_evm_forging_fee_payments.py

```diff
@@ -285,6 +316,49 @@ def run_test(self):
             assert_equal(forger_fees[i], api_fee_payments_node1[i - 1]['value'],
                          "Different fee value found for payment " + str(i))
 
+        # trigger cert submission
+        # Generate 2 SC blocks on SC node and start them automatic cert creation.
+        generate_next_block(sc_node_1, "second node")  # 1 MC block to reach the end of WE
+        generate_next_block(sc_node_1, "second node")  # 1 MC block to trigger Submitter logic
+
+        # Wait for Certificates appearance
+        time.sleep(10)
+        while mc_node.getmempoolinfo()["size"] < 1 and sc_node_1.submitter_isCertGenerationActive()["result"][
+            "state"]:
+            logging.info("Wait for certificates in the MC mempool...")
+            if sc_node_1.submitter_isCertGenerationActive()["result"]["state"]:
+                logging.info("sc_node generating certificate now.")
+            time.sleep(2)
+
+        assert_equal(1, mc_node.getmempoolinfo()["size"], "Certificates was not added to MC node mempool.")
+
+        # Advance to epoch 60 to enable forger pool fork. First block will already be counted for the distribution
+        # Generate more blocks so that in total there were 5 blocks from node_1 and 3 blocks from node_2
+        self.advance_to_epoch(60)
+        generate_next_blocks(sc_node_1, "first node", 4)
+        generate_next_blocks(sc_node_2, "second node", 2)
+
+        mc_node.generate(self.withdrawalEpochLength)
+        self.sc_sync_all()
+        generate_next_block(sc_node_2, "second node")
+        self.sc_sync_all()
+        per_block_fee = convertZenToWei(ft_pool_amount) // 8
+        node_1_fees = per_block_fee * 5
+        node_2_fees = per_block_fee * 3
+
+        sc_node_1_balance_before_payments = sc_node_1_balance_after_payments
+        sc_node_2_balance_before_payments = sc_node_2_balance_after_payments
+        sc_node_1_balance_after_payments = sc_node_1.wallet_getTotalBalance()['result']['balance']
+        sc_node_2_balance_after_payments = sc_node_2.wallet_getTotalBalance()['result']['balance']
+        assert_equal(sc_node_1_balance_before_payments + node_1_fees, sc_node_1_balance_after_payments,
+                     "Wrong fee payment amount for SC node 1")
+        assert_equal(sc_node_2_balance_before_payments + node_2_fees, sc_node_2_balance_after_payments,
+                     "Wrong fee payment amount for SC node 2")
+
+        # assert forger pool balance is 0 now, as the fees are distributed
+        forger_pool_balance = int(self.sc_nodes[0].rpc_eth_getBalance(format_evm(FORGER_POOL_RECIPIENT_ADDRESS), 'latest')['result'], 16)
+        assert_equal(forger_pool_balance, 0)
+
```

Comment:
we need to test (here or in a separate test) also this corner case please:
- withdrawal epoch N ends and fees are redistributed
- a revert on mainchain happens, causing the sidechain to go back to previous withdrawal epoch N-1
- balances are reverted correctly
- new  blocks are forgerd and we switch again withrawal epoch N
- fees are redistributed again correctly